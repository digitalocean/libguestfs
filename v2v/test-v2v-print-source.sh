#!/bin/bash -
# libguestfs virt-v2v test script
# Copyright (C) 2014 Red Hat Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

# Test --print-source option.

set -e

$TEST_FUNCTIONS
skip_if_skipped
skip_unless_phony_guest windows.img

d=test-v2v-print-source.d
rm -rf $d
mkdir $d

$VG virt-v2v --debug-gc \
    -i libvirtxml test-v2v-print-source.xml \
    -o local -os $d \
    --print-source > $d/output

mv $d/output $d/output.orig
< $d/output.orig \
grep -v 'Opening the source' |
grep -v 'Source guest information' |
sed -e 's,/.*/windows.img,windows.img,' |
grep -v '^$' \
> $d/output

diff -u test-v2v-print-source.expected $d/output

rm -r $d
