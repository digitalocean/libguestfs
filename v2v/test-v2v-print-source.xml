<domain type='kvm'>
  <name>windows</name>
  <memory>1048576</memory>
  <cpu match="minimum">
    <vendor>Intel</vendor>
    <model fallback="allow">Broadwell</model>
    <topology sockets="4" cores="8" threads="2"/>
  </cpu>
  <features>
    <acpi/>
    <apic/>
    <pae/>
  </features>
  <os>
    <type>hvm</type>
    <boot dev='hd'/>
  </os>
  <devices>
    <disk type='file' device='disk'>
      <driver name='qemu' type='raw'/>
      <source file='../test-data/phony-guests/windows.img'/>
      <target dev='vda' bus='virtio'/>
    </disk>
    <interface type='network'>
      <mac address='00:11:22:33:44:55'/>
      <source network='default'/>
      <model type='virtio'/>
    </interface>
    <video>
      <model type='qxl' ram='65536' vram='65536' vgamem='16384' heads='1'/>
    </video>
  </devices>
</domain>
