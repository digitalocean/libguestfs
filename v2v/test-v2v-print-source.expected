    source name: windows
hypervisor type: kvm
         memory: 1073741824 (bytes)
       nr vCPUs: 64
     CPU vendor: Intel
      CPU model: Broadwell
   CPU topology: sockets: 4 cores/socket: 8 threads/core: 2
   CPU features: pae,apic,acpi
       firmware: unknown
        display: 
          video: qxl
          sound: 
disks:
	windows.img (raw) [virtio-blk]
removable media:
NICs:
	Network "default" mac: 00:11:22:33:44:55 [virtio]
